src/Workspaces/Core/Portable/SolutionCrawler/CompilerDiagnosticsScope.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.SolutionCrawler
+{
+    internal enum CompilerDiagnosticsScope
+    {
+        // NOTE: Do not change existing field ordering/values as this scope is saved as a user option,
+        //       and that would break users who have saved a non-default scope for the option.
+
+        /// <summary>
+        /// Compiler warnings and errors are disabled for all documents.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Compiler warnings and errors are computed only for visible documents.
```

Comment:
doc that thsi also includes prior docs that had errors?

Reply:
Yes, will do.